### PR TITLE
15 adding user privileges

### LINKED ISSUE #15: Adding_user_privileges
create a new database with three tables, servers, users and privileges.
Stuck into how to link them together.

### PR DESCRIPTION
**Pull request by Yousifff**
rebuild db with three tables,  server,users,and privileges.

### CODE REVIEW

File: datastore/db.py [hunk 1]
```diff
@@ -1,21 +1,55 @@
...
+    def list_users(self):
+        users = self.cursor.execute("SELECT * FROM users;")
+        return users
+    
+    def user_login(self,username,password):
+        result = self.cursor.execute('SELECT * FROM users WHERE usernme=? AND password=?;',(username,password))
+        self.conn.commit()
+        return result.fetchone()
+    
+    def add_privilege(self,ip,username,stop,start):
```
Comment (Yousifff) [inline datastore/db.py:50]: not sure if this is the correct way to disable and enable streamlit components based on integer value stored in db.

File: datastore/db.py [hunk 2]
```diff
@@ -1,21 +1,55 @@
...
+        if stop:
+            self.cursor.execute("INSERT INTO permissions VALUES (?,?,?,?,?)",(ip,username,1,0,0))
+        else:
+            self.cursor.execute("INSERT INTO permissions VALUES (?,?,?,?,?)",(ip,username,0,0,1))
+        self.conn.commit()
```
Comment (Yousifff) [inline datastore/db.py]: Not sure of this the correct way to add privilege.
if for example stop privilege is 1 and start is 0 , start button must be disabled. 
so the user has permission to stop only.
I read stramlit documentation haven't find topic related to authorization .

Comment (Zaubeerer) [reply datastore/db.py]: I think we answered this question, right?

Let me know, if you have further questions :)

File: datastore/__pycache__/db.cpython-310.pyc
Comment (Yousifff) [inline datastore/__pycache__/db.cpython-310.pyc:1]: please remove the file form git.

**Review by Yousifff — COMMENTED** [review]
hey i made a first review, please make the following changes:

**Comment by Zaubeerer** [comment]
Hey @Yousifff, can you reopen this PR?

If not, maybe something went wrong when we attempted to fix the git history 🤔

**Pull request closed without merging**